packages/graphql/src/components/DeferredInterfaceValidation.ts

```diff
@@ -0,0 +1,249 @@
+import { Children, isRefkey, Refkey } from "@alloy-js/core";
+import { GraphQLOutputSymbol } from "../symbols/graphql-output-symbol.js";
+import { ref } from "../symbols/reference.js";
+
+interface FieldInfo {
+  name: string;
+  type: string;
+  args: Array<{ name: string; type: string }>;
+}
```

Comment:
In GraphQL, args are very similar to fields. Indeed, in this context the only difference is that args cannot have args.

Can we do
```typescript
interface FieldInfo {
  name: string;
  type: string;
}

interface ObjectFieldInfo extends FieldInfo {
  args: FieldInfo[];
}
```

as I feel like you have a number of functions that could work equivalently on any `FieldInfo` object rather than treating fields and args differently.